src/app/icd/ICDManager.cpp

```diff
@@ -233,5 +259,69 @@ void ICDManager::OnTransitionToIdle(System::Layer * aLayer, void * appState)
     pIcdManager->mStateObserver->OnTransitionToIdle();
 }
 
+bool ICDManager::EnsureChipStackLock()
+{
```

Comment:
Should `#if CHIP_STACK_LOCK_TRACKING_ENABLED` be used here?

Reply:
True, I will add it